Index: pulseaudio-0.9.16~test4/src/daemon/default.pa.in
===================================================================
--- pulseaudio-0.9.16~test4.orig/src/daemon/default.pa.in	2009-08-18 22:58:15.000000000 -0400
+++ pulseaudio-0.9.16~test4/src/daemon/default.pa.in	2009-08-18 22:58:49.000000000 -0400
@@ -29,6 +29,9 @@
 
 .fail
 
+### Automatically suspend sinks/sources that become idle for too long
+load-module module-suspend-on-idle
+
 ### Automatically restore the volume of streams and devices
 load-module module-device-restore
 load-module module-stream-restore
@@ -103,9 +106,6 @@
 ### Honour intended role device property
 load-module module-intended-roles
 
-### Automatically suspend sinks/sources that become idle for too long
-load-module module-suspend-on-idle
-
 ### If autoexit on idle is enabled we want to make sure we only quit
 ### when no local session needs us anymore.
 load-module module-console-kit
