Index: pulseaudio-0.99.1/src/daemon/default.pa.in
===================================================================
--- pulseaudio-0.99.1.orig/src/daemon/default.pa.in	2011-07-03 15:01:52.000000000 +0200
+++ pulseaudio-0.99.1/src/daemon/default.pa.in	2011-08-03 11:47:37.911239018 +0200
@@ -150,7 +150,7 @@
 load-module module-position-event-sounds
 
 ### Cork music streams when a phone stream is active
-load-module module-cork-music-on-phone
+# load-module module-cork-music-on-phone
 ])dnl
 
 ifelse(@HAVE_X11@, 1, [dnl
