Index: pulseaudio-0.9.19/src/daemon/esdcompat.in
===================================================================
--- pulseaudio-0.9.19.orig/src/daemon/esdcompat.in	2009-10-07 23:48:08.000000000 -0400
+++ pulseaudio-0.9.19/src/daemon/esdcompat.in	2009-10-08 00:01:28.000000000 -0400
@@ -93,4 +93,7 @@
     shift
 done
 
-eval "exec '@PA_BINARY@'$ARGS"
+if [ ! -f /var/lib/pulseaudio/pulse_a11y_nostart ] && \
+   [ ! -f $HOME/.pulse_a11y_nostart ]; then
+       eval "exec '@PA_BINARY@'$ARGS"
+fi
