Index: pulseaudio-0.9.13/src/pulsecore/sample-util.c
===================================================================
--- pulseaudio-0.9.13.orig/src/pulsecore/sample-util.c	2009-01-03 18:30:13.000000000 -0500
+++ pulseaudio-0.9.13/src/pulsecore/sample-util.c	2009-01-03 18:30:30.000000000 -0500
@@ -777,7 +777,7 @@
     return (l/fs) * fs;
 }
 
-int pa_frame_aligned(size_t l, const pa_sample_spec *ss) {
+pa_bool_t pa_frame_aligned(size_t l, const pa_sample_spec *ss) {
     size_t fs;
 
     pa_assert(ss);
Index: pulseaudio-0.9.13/src/pulsecore/sample-util.h
===================================================================
--- pulseaudio-0.9.13.orig/src/pulsecore/sample-util.h	2009-01-03 18:30:13.000000000 -0500
+++ pulseaudio-0.9.13/src/pulsecore/sample-util.h	2009-01-03 18:30:30.000000000 -0500
@@ -71,7 +71,7 @@
 
 size_t pa_frame_align(size_t l, const pa_sample_spec *ss) PA_GCC_PURE;
 
-int pa_frame_aligned(size_t l, const pa_sample_spec *ss) PA_GCC_PURE;
+pa_bool_t pa_frame_aligned(size_t l, const pa_sample_spec *ss) PA_GCC_PURE;
 
 void pa_interleave(const void *src[], unsigned channels, void *dst, size_t ss, unsigned n);
 void pa_deinterleave(const void *src, void *dst[], unsigned channels, size_t ss, unsigned n);
