Index: pulseaudio-0.9.22/src/modules/alsa/mixer/paths/analog-input.conf.common
===================================================================
--- pulseaudio-0.9.22.orig/src/modules/alsa/mixer/paths/analog-input.conf.common	2010-11-25 19:45:22.000000000 -0500
+++ pulseaudio-0.9.22/src/modules/alsa/mixer/paths/analog-input.conf.common	2010-12-18 08:42:33.000000000 -0500
@@ -66,6 +66,10 @@
 name = input-microphone
 priority = 20
 
+[Option Input Source:Digital Mic]
+name = input-microphone
+priority = 20
+
 [Option Input Source:Mic]
 name = input-microphone
 priority = 20
