Index: pulseaudio-0.9.13/src/daemon/default.pa.in
===================================================================
--- pulseaudio-0.9.13.orig/src/daemon/default.pa.in	2008-12-28 23:28:08.000000000 -0500
+++ pulseaudio-0.9.13/src/daemon/default.pa.in	2008-12-28 23:30:50.000000000 -0500
@@ -29,6 +29,10 @@
 
 .fail
 
+### Automatically restore the volume of streams and devices
+load-module module-stream-restore
+load-module module-device-restore
+
 ### Load audio drivers statically (it's probably better to not load
 ### these drivers manually, but instead use module-hal-detect --
 ### see below -- for doing this automatically)
@@ -81,11 +85,6 @@
 .fail
 .endif
 
-
-### Automatically restore the volume of streams and devices
-load-module module-stream-restore
-load-module module-device-restore
-
 ### Automatically restore the default sink/source when changed by the user during runtime
 load-module module-default-device-restore
 
