From e5a6af86fa3463b186db528c259ad76d97d7110f Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Tue, 30 Nov 2010 22:11:48 +0100
Subject: [PATCH 2/2] JACK: Load module-jackdbus-detect in default.pa

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 src/daemon/default.pa.in |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index 00c000e..e896385 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -62,6 +62,11 @@ load-module module-detect
 load-module module-bluetooth-discover
 .endif
 
+### Automatically connect sink and source if JACK server is present
+.ifexists module-jackdbus-detect@PA_SOEXT@
+load-module module-jackdbus-detect
+.endif
+
 ### Load several protocols
 .ifexists module-esound-protocol-unix@PA_SOEXT@
 load-module module-esound-protocol-unix
-- 
1.7.1

