From e5a6af86fa3463b186db528c259ad76d97d7110f Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Tue, 30 Nov 2010 22:11:48 +0100
Subject: [PATCH 2/2] JACK: Load module-jackdbus-detect in default.pa

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 src/daemon/default.pa.in |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

Index: pulseaudio-0.99.1/src/daemon/default.pa.in
===================================================================
--- pulseaudio-0.99.1.orig/src/daemon/default.pa.in	2011-08-03 11:56:16.000000000 +0200
+++ pulseaudio-0.99.1/src/daemon/default.pa.in	2011-08-03 11:58:45.081237445 +0200
@@ -79,6 +79,11 @@
 load-module module-detect
 .endif
 
+### Automatically connect sink and source if JACK server is present
+.ifexists module-jackdbus-detect@PA_SOEXT@
+load-module module-jackdbus-detect
+.endif
+
 ifelse(@HAVE_BLUEZ@, 1, [dnl
 ### Automatically load driver modules for Bluetooth hardware
 .ifexists module-bluetooth-discover@PA_SOEXT@
