Index: pulseaudio-0.9.14/src/modules/module-alsa-sink.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/module-alsa-sink.c	2009-01-31 15:13:06.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/module-alsa-sink.c	2009-01-31 15:13:45.000000000 -0500
@@ -615,7 +615,7 @@
         pa_usec_t sleep_usec, process_usec;
 
         hw_sleep_time(u, &sleep_usec, &process_usec);
-        avail_min += pa_usec_to_bytes(sleep_usec, &u->sink->sample_spec);
+        avail_min += pa_usec_to_bytes(sleep_usec, &u->sink->sample_spec) / u->frame_size;
     }
 
     pa_log_debug("setting avail_min=%lu", (unsigned long) avail_min);
Index: pulseaudio-0.9.14/src/modules/module-alsa-source.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/module-alsa-source.c	2009-01-31 15:13:06.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/module-alsa-source.c	2009-01-31 15:14:52.000000000 -0500
@@ -560,7 +560,7 @@
         pa_usec_t sleep_usec, process_usec;
 
         hw_sleep_time(u, &sleep_usec, &process_usec);
-        avail_min += pa_usec_to_bytes(sleep_usec, &u->source->sample_spec);
+        avail_min += pa_usec_to_bytes(sleep_usec, &u->source->sample_spec) / u->frame_size;
     }
 
     pa_log_debug("setting avail_min=%lu", (unsigned long) avail_min);
