Index: pulseaudio-0.9.14/src/modules/bluetooth/module-bluetooth-device.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/bluetooth/module-bluetooth-device.c	2009-01-31 15:29:51.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/bluetooth/module-bluetooth-device.c	2009-01-31 15:30:04.000000000 -0500
@@ -448,8 +448,11 @@
         }
         u->ss.format = PA_SAMPLE_S16LE;
     }
-    else
-        u->ss.format = PA_SAMPLE_U8;
+    else {
+        u->ss.format = PA_SAMPLE_S16LE;
+        u->ss.channels = 1;
+        u->ss.rate = 8000;
+    }
 
     memset(msg.buf, 0, BT_SUGGESTED_BUFFER_SIZE);
     msg.setconf_req.h.type = BT_REQUEST;
