Index: pulseaudio-2.99.1/src/daemon/default.pa.in
===================================================================
--- pulseaudio-2.99.1.orig/src/daemon/default.pa.in	2012-11-16 11:41:51.136239688 +0100
+++ pulseaudio-2.99.1/src/daemon/default.pa.in	2012-11-16 11:41:54.600239637 +0100
@@ -166,9 +166,9 @@
 
 ifelse(@HAVE_DBUS@, 1, [dnl
 ### Load DBus protocol
-.ifexists module-dbus-protocol@PA_SOEXT@
-load-module module-dbus-protocol
-.endif
+#.ifexists module-dbus-protocol@PA_SOEXT@
+#load-module module-dbus-protocol
+#.endif
 ])dnl
 
 ifelse(@HAVE_X11@, 1, [dnl
