Index: pulseaudio-0.9.13/src/pulsecore/macro.h
===================================================================
--- pulseaudio-0.9.13.orig/src/pulsecore/macro.h	2009-01-03 18:39:07.000000000 -0500
+++ pulseaudio-0.9.13/src/pulsecore/macro.h	2009-01-03 18:39:18.000000000 -0500
@@ -40,7 +40,7 @@
 #ifndef PA_LIKELY
 #ifdef __GNUC__
 #define PA_LIKELY(x) (__builtin_expect(!!(x),1))
-#define PA_UNLIKELY(x) (__builtin_expect((x),0))
+#define PA_UNLIKELY(x) (__builtin_expect(!!(x),0))
 #else
 #define PA_LIKELY(x) (x)
 #define PA_UNLIKELY(x) (x)
