Index: pulseaudio-0.9.14/src/modules/alsa-util.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/alsa-util.c	2009-01-31 15:54:00.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/alsa-util.c	2009-01-31 15:55:22.000000000 -0500
@@ -725,7 +725,10 @@
             return NULL;
         }
 
-        *dev = d;
+        if (dev)
+            *dev = d;
+        else
+            pa_xfree(d);
 
         if (ss->channels != map->channels)
             pa_channel_map_init_extend(map, ss->channels, PA_CHANNEL_MAP_ALSA);
