Index: pulseaudio-0.9.14/src/pulse/volume.c
===================================================================
--- pulseaudio-0.9.14.orig/src/pulse/volume.c	2009-01-31 16:22:02.000000000 -0500
+++ pulseaudio-0.9.14/src/pulse/volume.c	2009-01-31 16:22:50.000000000 -0500
@@ -304,7 +304,7 @@
         p == PA_CHANNEL_POSITION_LFE;
 }
 
-pa_cvolume *pa_cvolume_remap(pa_cvolume *v, pa_channel_map *from, pa_channel_map *to) {
+pa_cvolume *pa_cvolume_remap(pa_cvolume *v, const pa_channel_map *from, const pa_channel_map *to) {
     int a, b;
     pa_cvolume result;
 
Index: pulseaudio-0.9.14/src/pulse/volume.h
===================================================================
--- pulseaudio-0.9.14.orig/src/pulse/volume.h	2009-01-31 16:22:02.000000000 -0500
+++ pulseaudio-0.9.14/src/pulse/volume.h	2009-01-31 16:23:21.000000000 -0500
@@ -207,7 +207,7 @@
 #endif
 
 /** Remap a volume from one channel mapping to a different channel mapping. \since 0.9.12 */
-pa_cvolume *pa_cvolume_remap(pa_cvolume *v, pa_channel_map *from, pa_channel_map *to);
+pa_cvolume *pa_cvolume_remap(pa_cvolume *v, const pa_channel_map *from, const pa_channel_map *to);
 
 /** Return non-zero if the specified volume is compatible with
  * the specified sample spec. \since 0.9.13 */
