Index: pulseaudio-0.9.14/src/modules/module-esound-sink.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/module-esound-sink.c	2009-01-31 16:25:18.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/module-esound-sink.c	2009-01-31 16:25:40.000000000 -0500
@@ -168,7 +168,7 @@
             w = pa_bytes_to_usec((uint64_t) u->offset + u->memchunk.length, &u->sink->sample_spec);
 
             *((pa_usec_t*) data) = w > r ? w - r : 0;
-            break;
+            return 0;
         }
 
         case SINK_MESSAGE_PASS_SOCKET: {
