Index: pulseaudio-0.9.13/src/modules/module-device-restore.c
===================================================================
--- pulseaudio-0.9.13.orig/src/modules/module-device-restore.c	2009-01-03 19:38:47.000000000 -0500
+++ pulseaudio-0.9.13/src/modules/module-device-restore.c	2009-01-03 19:40:30.000000000 -0500
@@ -332,7 +332,7 @@
     if (!fname)
         goto fail;
 
-    if (!(u->gdbm_file = gdbm_open(fname, 0, GDBM_WRCREAT, 0600, NULL))) {
+    if (!(u->gdbm_file = gdbm_open(fname, 0, GDBM_WRCREAT|GDBM_NOLOCK, 0600, NULL))) {
         pa_log("Failed to open volume database '%s': %s", fname, gdbm_strerror(gdbm_errno));
         pa_xfree(fname);
         goto fail;
Index: pulseaudio-0.9.13/src/modules/module-stream-restore.c
===================================================================
--- pulseaudio-0.9.13.orig/src/modules/module-stream-restore.c	2009-01-03 19:38:55.000000000 -0500
+++ pulseaudio-0.9.13/src/modules/module-stream-restore.c	2009-01-03 19:40:30.000000000 -0500
@@ -741,7 +741,7 @@
     if (!fname)
         goto fail;
 
-    if (!(u->gdbm_file = gdbm_open(fname, 0, GDBM_WRCREAT, 0600, NULL))) {
+    if (!(u->gdbm_file = gdbm_open(fname, 0, GDBM_WRCREAT|GDBM_NOLOCK, 0600, NULL))) {
         pa_log("Failed to open volume database '%s': %s", fname, gdbm_strerror(gdbm_errno));
         pa_xfree(fname);
         goto fail;
