Index: pulseaudio-0.9.14/src/pulsecore/sink.c
===================================================================
--- pulseaudio-0.9.14.orig/src/pulsecore/sink.c	2009-02-04 19:21:36.000000000 -0500
+++ pulseaudio-0.9.14/src/pulsecore/sink.c	2009-02-04 19:21:54.000000000 -0500
@@ -1235,6 +1235,10 @@
         case PA_SINK_MESSAGE_SET_STATE:
 
             s->thread_info.state = PA_PTR_TO_UINT(userdata);
+
+            if (s->thread_info.state == PA_SINK_SUSPENDED)
+                s->thread_info.rewind_requested = FALSE;
+
             return 0;
 
         case PA_SINK_MESSAGE_DETACH:
