Index: pulseaudio-0.9.13/src/modules/rtp/module-rtp-send.c
===================================================================
--- pulseaudio-0.9.13.orig/src/modules/rtp/module-rtp-send.c	2009-01-07 17:21:57.000000000 -0500
+++ pulseaudio-0.9.13/src/modules/rtp/module-rtp-send.c	2009-01-07 17:22:11.000000000 -0500
@@ -296,6 +296,11 @@
             pa_log("IP_MULTICAST_TTL failed: %s", pa_cstrerror(errno));
             goto fail;
         }
+
+        if (setsockopt(sap_fd, IPPROTO_IP, IP_MULTICAST_TTL, &_ttl, sizeof(_ttl)) < 0) {
+            pa_log("IP_MULTICAST_TTL (sap) failed: %s", pa_cstrerror(errno));
+            goto fail;
+        }
     }
 
     /* If the socket queue is full, let's drop packets */
