Index: pulseaudio-0.9.14/src/modules/module-suspend-on-idle.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/module-suspend-on-idle.c	2009-02-12 18:26:45.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/module-suspend-on-idle.c	2009-02-12 18:28:17.000000000 -0500
@@ -172,6 +172,9 @@
     pa_source_output_assert_ref(s);
     pa_assert(u);
 
+    if (!s->source)
+        return PA_HOOK_OK;
+
     if (pa_source_used_by(s->source) <= 0) {
         struct device_info *d;
         if ((d = pa_hashmap_get(u->device_infos, s->source)))
