Index: pulseaudio/src/daemon/default.pa.in
===================================================================
--- pulseaudio.orig/src/daemon/default.pa.in
+++ pulseaudio/src/daemon/default.pa.in
@@ -164,3 +164,8 @@ load-module module-filter-apply
 ### Make some devices default
 #set-default-sink output
 #set-default-source input
+
+### Use hot-plugged devices like Bluetooth or USB automatically (LP: #1702794)
+.ifexists module-switch-on-connect.so
+load-module module-switch-on-connect
+.endif
