Index: pulseaudio-0.9.14/src/pulse/scache.c
===================================================================
--- pulseaudio-0.9.14.orig/src/pulse/scache.c	2009-02-13 22:58:05.000000000 -0500
+++ pulseaudio-0.9.14/src/pulse/scache.c	2009-02-13 22:58:26.000000000 -0500
@@ -66,10 +66,8 @@
     pa_tagstruct_put_channel_map(t, &s->channel_map);
     pa_tagstruct_putu32(t, (uint32_t) length);
 
-    if (s->context->version >= 13) {
-        pa_init_proplist(s->proplist);
+    if (s->context->version >= 13)
         pa_tagstruct_put_proplist(t, s->proplist);
-    }
 
     pa_pstream_send_tagstruct(s->context->pstream, t);
     pa_pdispatch_register_reply(s->context->pdispatch, tag, DEFAULT_TIMEOUT, pa_create_stream_callback, s, NULL);
