Index: pulseaudio-0.9.14/src/pulsecore/sink.c
===================================================================
--- pulseaudio-0.9.14.orig/src/pulsecore/sink.c	2009-02-19 18:45:44.000000000 -0500
+++ pulseaudio-0.9.14/src/pulsecore/sink.c	2009-02-19 18:46:04.000000000 -0500
@@ -1557,6 +1557,9 @@
 
     pa_sink_assert_ref(s);
 
+    pa_assert(!min_latency || !max_latency ||
+              min_latency <= max_latency);
+
     s->thread_info.min_latency = min_latency;
     s->thread_info.max_latency = max_latency;
 
Index: pulseaudio-0.9.14/src/pulsecore/source.c
===================================================================
--- pulseaudio-0.9.14.orig/src/pulsecore/source.c	2009-02-19 18:46:57.000000000 -0500
+++ pulseaudio-0.9.14/src/pulsecore/source.c	2009-02-19 18:47:08.000000000 -0500
@@ -994,6 +994,9 @@
 
     pa_source_assert_ref(s);
 
+    pa_assert(!min_latency || !max_latency ||
+              min_latency <= max_latency);
+
     s->thread_info.min_latency = min_latency;
     s->thread_info.max_latency = max_latency;
 
