Index: pulseaudio-0.9.16~test5/src/modules/alsa/mixer/paths/analog-output-lfe-on-mono.conf
===================================================================
--- pulseaudio-0.9.16~test5.orig/src/modules/alsa/mixer/paths/analog-output-lfe-on-mono.conf	2009-08-21 00:22:24.000000000 -0400
+++ pulseaudio-0.9.16~test5/src/modules/alsa/mixer/paths/analog-output-lfe-on-mono.conf	2009-08-21 00:22:41.000000000 -0400
@@ -41,9 +41,12 @@
 override-map.1 = lfe
 override-map.2 = lfe,lfe
 
+; This profile path is intended to control the speaker, not the
+; headphones. But it should not hurt if we leave the headphone jack
+; enabled nonetheless.
 [Element Headphone]
-switch = off
-volume = off
+switch = mute
+volume = zero
 
 [Element Speaker]
 switch = mute
Index: pulseaudio-0.9.16~test5/src/modules/alsa/mixer/paths/analog-output-mono.conf
===================================================================
--- pulseaudio-0.9.16~test5.orig/src/modules/alsa/mixer/paths/analog-output-mono.conf	2009-08-21 00:22:25.000000000 -0400
+++ pulseaudio-0.9.16~test5/src/modules/alsa/mixer/paths/analog-output-mono.conf	2009-08-21 00:22:41.000000000 -0400
@@ -38,9 +38,12 @@
 override-map.1 = all
 override-map.2 = all-left,all-right
 
+; This profile path is intended to control the speaker, not the
+; headphones. But it should not hurt if we leave the headphone jack
+; enabled nonetheless.
 [Element Headphone]
-switch = off
-volume = off
+switch = mute
+volume = zero
 
 [Element Speaker]
 switch = mute
Index: pulseaudio-0.9.16~test5/src/modules/alsa/mixer/paths/analog-output.conf
===================================================================
--- pulseaudio-0.9.16~test5.orig/src/modules/alsa/mixer/paths/analog-output.conf	2009-08-21 00:22:25.000000000 -0400
+++ pulseaudio-0.9.16~test5/src/modules/alsa/mixer/paths/analog-output.conf	2009-08-21 00:22:41.000000000 -0400
@@ -37,9 +37,12 @@
 switch = off
 volume = off
 
+; This profile path is intended to control the speaker, not the
+; headphones. But it should not hurt if we leave the headphone jack
+; enabled nonetheless.
 [Element Headphone]
-switch = off
-volume = off
+switch = mute
+volume = zero
 
 [Element Speaker]
 switch = mute
