Index: pulseaudio-0.9.9/src/modules/alsa-util.c
===================================================================
--- pulseaudio-0.9.9.orig/src/modules/alsa-util.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/modules/alsa-util.c	2008-02-18 18:40:06.000000000 -0500
@@ -615,8 +615,10 @@
 
         *dev = d;
 
-        if (ss->channels != map->channels)
+        if (ss->channels != map->channels) {
+            pa_assert_se(pa_channel_map_init_auto(map, ss->channels, PA_CHANNEL_MAP_AUX));
             pa_channel_map_init_auto(map, ss->channels, PA_CHANNEL_MAP_ALSA);
+        }
 
         return pcm_handle;
     }
Index: pulseaudio-0.9.9/src/modules/module-combine.c
===================================================================
--- pulseaudio-0.9.9.orig/src/modules/module-combine.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/modules/module-combine.c	2008-02-18 18:40:06.000000000 -0500
@@ -988,8 +988,10 @@
 
     if (master_sink && ss.channels == master_sink->sample_spec.channels)
         map = master_sink->channel_map;
-    else
+    else {
+        pa_assert_se(pa_channel_map_init_auto(&map, ss.channels, PA_CHANNEL_MAP_AUX));
         pa_channel_map_init_auto(&map, ss.channels, PA_CHANNEL_MAP_DEFAULT);
+    }
 
     if ((pa_modargs_get_channel_map(ma, NULL, &map) < 0)) {
         pa_log("Invalid channel map.");
Index: pulseaudio-0.9.9/src/modules/module-jack-sink.c
===================================================================
--- pulseaudio-0.9.9.orig/src/modules/module-jack-sink.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/modules/module-jack-sink.c	2008-02-18 18:40:06.000000000 -0500
@@ -333,6 +333,7 @@
         goto fail;
     }
 
+    pa_assert_se(pa_channel_map_init_auto(&map, channels, PA_CHANNEL_MAP_AUX));
     pa_channel_map_init_auto(&map, channels, PA_CHANNEL_MAP_ALSA);
     if (pa_modargs_get_channel_map(ma, NULL, &map) < 0 || map.channels != channels) {
         pa_log("Failed to parse channel_map= argument.");
Index: pulseaudio-0.9.9/src/modules/module-jack-source.c
===================================================================
--- pulseaudio-0.9.9.orig/src/modules/module-jack-source.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/modules/module-jack-source.c	2008-02-18 18:40:06.000000000 -0500
@@ -304,6 +304,7 @@
         goto fail;
     }
 
+    pa_assert_se(pa_channel_map_init_auto(&map, channels, PA_CHANNEL_MAP_AUX));
     pa_channel_map_init_auto(&map, channels, PA_CHANNEL_MAP_ALSA);
     if (pa_modargs_get_channel_map(ma, NULL, &map) < 0 || map.channels != channels) {
         pa_log("failed to parse channel_map= argument.");
Index: pulseaudio-0.9.9/src/modules/module-zeroconf-discover.c
===================================================================
--- pulseaudio-0.9.9.orig/src/modules/module-zeroconf-discover.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/modules/module-zeroconf-discover.c	2008-02-18 18:40:06.000000000 -0500
@@ -164,6 +164,7 @@
         pa_module *m;
 
         ss = u->core->default_sample_spec;
+        pa_assert_se(pa_channel_map_init_auto(&cm, ss.channels, PA_CHANNEL_MAP_AUX));
         pa_channel_map_init_auto(&cm, ss.channels, PA_CHANNEL_MAP_DEFAULT);
 
         for (l = txt; l; l = l->next) {
@@ -189,8 +190,10 @@
             avahi_free(value);
         }
 
-        if (!channel_map_set && cm.channels != ss.channels)
+        if (!channel_map_set && cm.channels != ss.channels) {
+            pa_assert_se(pa_channel_map_init_auto(&cm, ss.channels, PA_CHANNEL_MAP_AUX));
             pa_channel_map_init_auto(&cm, ss.channels, PA_CHANNEL_MAP_DEFAULT);
+        }
 
         if (!pa_sample_spec_valid(&ss)) {
             pa_log("Service '%s' contains an invalid sample specification.", name);
Index: pulseaudio-0.9.9/src/pulse/stream.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulse/stream.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/pulse/stream.c	2008-02-18 18:40:06.000000000 -0500
@@ -46,6 +46,7 @@
 pa_stream *pa_stream_new(pa_context *c, const char *name, const pa_sample_spec *ss, const pa_channel_map *map) {
     pa_stream *s;
     int i;
+    pa_channel_map tmap;
 
     pa_assert(c);
     pa_assert(PA_REFCNT_VALUE(c) >= 1);
@@ -54,6 +55,9 @@
     PA_CHECK_VALIDITY_RETURN_NULL(c, c->version >= 12 || (ss->format != PA_SAMPLE_S32LE || ss->format != PA_SAMPLE_S32NE), PA_ERR_NOTSUPPORTED);
     PA_CHECK_VALIDITY_RETURN_NULL(c, !map || (pa_channel_map_valid(map) && map->channels == ss->channels), PA_ERR_INVALID);
 
+    if (!map)
+        PA_CHECK_VALIDITY_RETURN_NULL(c, map = pa_channel_map_init_auto(&tmap, ss->channels, PA_CHANNEL_MAP_DEFAULT), PA_ERR_INVALID);
+
     s = pa_xnew(pa_stream, 1);
     PA_REFCNT_INIT(s);
     s->context = c;
@@ -81,13 +85,9 @@
     s->direction = PA_STREAM_NODIRECTION;
     s->name = pa_xstrdup(name);
     s->sample_spec = *ss;
+    s->channel_map = *map;
     s->flags = 0;
 
-    if (map)
-        s->channel_map = *map;
-    else
-        pa_channel_map_init_auto(&s->channel_map, ss->channels, PA_CHANNEL_MAP_DEFAULT);
-
     s->channel = 0;
     s->channel_valid = 0;
     s->syncid = c->csyncid++;
Index: pulseaudio-0.9.9/src/pulsecore/core-scache.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/core-scache.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/core-scache.c	2008-02-18 18:40:06.000000000 -0500
@@ -145,9 +145,16 @@
 int pa_scache_add_item(pa_core *c, const char *name, const pa_sample_spec *ss, const pa_channel_map *map, const pa_memchunk *chunk, uint32_t *idx) {
     pa_scache_entry *e;
     char st[PA_SAMPLE_SPEC_SNPRINT_MAX];
+    pa_channel_map tmap;
 
     pa_assert(c);
     pa_assert(name);
+    pa_assert(!ss || pa_sample_spec_valid(ss));
+    pa_assert(!map || (pa_channel_map_valid(map) && ss && ss->channels == map->channels));
+
+    if (ss && !map)
+        if (!(map = pa_channel_map_init_auto(&tmap, ss->channels, PA_CHANNEL_MAP_DEFAULT)))
+            return -1;
 
     if (chunk && chunk->length > PA_SCACHE_ENTRY_SIZE_MAX)
         return -1;
@@ -155,9 +162,11 @@
     if (!(e = scache_add_item(c, name)))
         return -1;
 
+    memset(&e->sample_spec, 0, sizeof(e->sample_spec));
+    pa_channel_map_init(&e->channel_map);
+
     if (ss) {
         e->sample_spec = *ss;
-        pa_channel_map_init_auto(&e->channel_map, ss->channels, PA_CHANNEL_MAP_DEFAULT);
         e->volume.channels = e->sample_spec.channels;
     }
 
Index: pulseaudio-0.9.9/src/pulsecore/resampler.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/resampler.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/resampler.c	2008-02-18 18:40:06.000000000 -0500
@@ -215,13 +215,13 @@
 
     if (am)
         r->i_cm = *am;
-    else
-        pa_channel_map_init_auto(&r->i_cm, r->i_ss.channels, PA_CHANNEL_MAP_DEFAULT);
+    else if (!pa_channel_map_init_auto(&r->i_cm, r->i_ss.channels, PA_CHANNEL_MAP_DEFAULT))
+        goto fail;
 
     if (bm)
         r->o_cm = *bm;
-    else
-        pa_channel_map_init_auto(&r->o_cm, r->o_ss.channels, PA_CHANNEL_MAP_DEFAULT);
+    else if (!pa_channel_map_init_auto(&r->o_cm, r->o_ss.channels, PA_CHANNEL_MAP_DEFAULT))
+        goto fail;
 
     r->i_fz = pa_frame_size(a);
     r->o_fz = pa_frame_size(b);
Index: pulseaudio-0.9.9/src/pulsecore/sink-input.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/sink-input.c	2008-02-18 18:39:47.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/sink-input.c	2008-02-18 18:40:06.000000000 -0500
@@ -120,7 +120,7 @@
         if (data->sink->channel_map.channels == data->sample_spec.channels)
             data->channel_map = data->sink->channel_map;
         else
-            pa_channel_map_init_auto(&data->channel_map, data->sample_spec.channels, PA_CHANNEL_MAP_DEFAULT);
+            pa_return_null_if_fail(pa_channel_map_init_auto(&data->channel_map, data->sample_spec.channels, PA_CHANNEL_MAP_DEFAULT));
     }
 
     pa_return_null_if_fail(pa_channel_map_valid(&data->channel_map));
Index: pulseaudio-0.9.9/src/pulsecore/sink.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/sink.c	2008-02-18 18:39:48.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/sink.c	2008-02-18 18:40:06.000000000 -0500
@@ -73,7 +73,7 @@
     pa_return_null_if_fail(pa_sample_spec_valid(spec));
 
     if (!map)
-        map = pa_channel_map_init_auto(&tmap, spec->channels, PA_CHANNEL_MAP_DEFAULT);
+        pa_return_null_if_fail((map = pa_channel_map_init_auto(&tmap, spec->channels, PA_CHANNEL_MAP_DEFAULT)));
 
     pa_return_null_if_fail(map && pa_channel_map_valid(map));
     pa_return_null_if_fail(map->channels == spec->channels);
Index: pulseaudio-0.9.9/src/pulsecore/sound-file.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/sound-file.c	2008-02-18 18:39:48.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/sound-file.c	2008-02-18 18:40:06.000000000 -0500
@@ -119,7 +119,10 @@
     }
 
     if (map)
-        pa_channel_map_init_auto(map, ss->channels, PA_CHANNEL_MAP_DEFAULT);
+        if (!pa_channel_map_init_auto(map, ss->channels, PA_CHANNEL_MAP_DEFAULT)) {
+            pa_log("Unsupported channel map in file %s", fname);
+            goto finish;
+        }
 
     if ((l = pa_frame_size(ss) * sfinfo.frames) > PA_SCACHE_ENTRY_SIZE_MAX) {
         pa_log("File too large");
Index: pulseaudio-0.9.9/src/pulsecore/source-output.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/source-output.c	2008-02-18 18:39:48.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/source-output.c	2008-02-18 18:40:06.000000000 -0500
@@ -97,7 +97,7 @@
         if (data->source->channel_map.channels == data->sample_spec.channels)
             data->channel_map = data->source->channel_map;
         else
-            pa_channel_map_init_auto(&data->channel_map, data->sample_spec.channels, PA_CHANNEL_MAP_DEFAULT);
+            pa_return_null_if_fail(pa_channel_map_init_auto(&data->channel_map, data->sample_spec.channels, PA_CHANNEL_MAP_DEFAULT));
     }
 
     pa_return_null_if_fail(pa_channel_map_valid(&data->channel_map));
Index: pulseaudio-0.9.9/src/pulsecore/source.c
===================================================================
--- pulseaudio-0.9.9.orig/src/pulsecore/source.c	2008-02-18 18:39:48.000000000 -0500
+++ pulseaudio-0.9.9/src/pulsecore/source.c	2008-02-18 18:40:06.000000000 -0500
@@ -64,7 +64,7 @@
     pa_return_null_if_fail(pa_sample_spec_valid(spec));
 
     if (!map)
-        map = pa_channel_map_init_auto(&tmap, spec->channels, PA_CHANNEL_MAP_DEFAULT);
+        pa_return_null_if_fail(map = pa_channel_map_init_auto(&tmap, spec->channels, PA_CHANNEL_MAP_DEFAULT));
 
     pa_return_null_if_fail(map && pa_channel_map_valid(map));
     pa_return_null_if_fail(map->channels == spec->channels);
