Index: pulseaudio-0.9.18/src/modules/bluetooth/module-bluetooth-device.c
===================================================================
--- pulseaudio-0.9.18.orig/src/modules/bluetooth/module-bluetooth-device.c	2009-09-28 23:34:55.000000000 -0400
+++ pulseaudio-0.9.18/src/modules/bluetooth/module-bluetooth-device.c	2009-09-28 23:58:25.000000000 -0400
@@ -2265,6 +2265,7 @@
 
 /* Run from main thread */
 static const pa_bluetooth_device* find_device(struct userdata *u, const char *address, const char *path) {
+    char *t = NULL;
     const pa_bluetooth_device *d = NULL;
 
     pa_assert(u);
@@ -2280,10 +2281,13 @@
             return NULL;
         }
 
-        if (address && !(pa_streq(d->address, address))) {
-            pa_log_error("Passed path %s and address %s don't match.", path, address);
+        t = pa_sprintf_malloc("%s", address);
+        if (t && !(pa_streq(d->address, t))) {
+            pa_log_error("Passed path %s and address %s don't match.", path, t);
+            pa_xfree(t);
             return NULL;
         }
+        pa_xfree(t);
 
     } else {
         if (!(d = pa_bluetooth_discovery_get_by_address(u->discovery, address))) {
