Index: pulseaudio-0.9.9/src/daemon/main.c
===================================================================
--- pulseaudio-0.9.9.orig/src/daemon/main.c	2008-02-18 19:01:04.000000000 -0500
+++ pulseaudio-0.9.9/src/daemon/main.c	2008-02-18 19:01:17.000000000 -0500
@@ -333,6 +333,7 @@
     int valid_pid_file = 0;
     gid_t gid = (gid_t) -1;
     pa_bool_t allow_realtime, allow_high_priority;
+    pa_bool_t ltdl_init = FALSE;
 
 #ifdef OS_IS_WIN32
     pa_time_event *timer;
@@ -505,6 +506,7 @@
 
     LTDL_SET_PRELOADED_SYMBOLS();
     pa_ltdl_init();
+    ltdl_init = TRUE;
 
     if (conf->dl_search_path)
         lt_dlsetsearchpath(conf->dl_search_path);
@@ -837,7 +839,8 @@
     WSACleanup();
 #endif
 
-    pa_ltdl_done();
+    if (ltdl_init)
+        pa_ltdl_done();
 
 #ifdef HAVE_DBUS
     dbus_shutdown();
