Index: pulseaudio-0.9.19-237-g721e/src/modules/alsa/mixer/paths/analog-output.conf.common
===================================================================
--- pulseaudio-0.9.19-237-g721e.orig/src/modules/alsa/mixer/paths/analog-output.conf.common	2009-11-06 19:20:10.000000000 -0500
+++ pulseaudio-0.9.19-237-g721e/src/modules/alsa/mixer/paths/analog-output.conf.common	2009-11-06 19:22:01.000000000 -0500
@@ -110,6 +110,9 @@
 name = output-amplifier-off
 priority = 0
 
+[Element IEC958 Optical Raw]
+switch = off
+
 ;;; 'Analog Output'
 
 [Element Analog Output]
