--- pulseaudio-0.9.9.orig/src/pulsecore/protocol-native.c	2008-02-23 14:28:31.000000000 +0100
+++ pulseaudio-0.9.9/src/pulsecore/protocol-native.c	2008-02-23 15:13:11.000000000 +0100
@@ -1177,6 +1177,7 @@
     pa_tagstruct_putu32(t, s->index);
     pa_tagstruct_putu32(t, o->source->index);
     pa_tagstruct_puts(t, o->source->name);
+    pa_tagstruct_put_boolean(t, pa_source_get_state(o->source) == PA_SOURCE_SUSPENDED);
     pa_pstream_send_tagstruct(s->connection->pstream, t);
 }
 
