Index: pulseaudio-0.9.21/src/modules/module-udev-detect.c
===================================================================
--- pulseaudio-0.9.21.orig/src/modules/module-udev-detect.c	2009-11-26 17:23:42.000000000 -0500
+++ pulseaudio-0.9.21/src/modules/module-udev-detect.c	2009-11-26 17:24:35.000000000 -0500
@@ -366,12 +366,6 @@
         return;
     }
 
-    if ((ff = udev_device_get_property_value(dev, "SOUND_FORM_FACTOR")) &&
-        pa_streq(ff, "modem")) {
-        pa_log_debug("Ignoring %s, because it is a modem.", udev_device_get_devpath(dev));
-        return;
-    }
-
     action = udev_device_get_action(dev);
 
     if (action && pa_streq(action, "remove"))
