Index: pulseaudio-0.9.22~0.9.21+stable-queue-32-g8478/src/daemon/start-pulseaudio-x11.in
===================================================================
--- pulseaudio-0.9.22~0.9.21+stable-queue-32-g8478.orig/src/daemon/start-pulseaudio-x11.in	2010-01-28 19:41:29.000000000 -0500
+++ pulseaudio-0.9.22~0.9.21+stable-queue-32-g8478/src/daemon/start-pulseaudio-x11.in	2010-01-28 19:44:55.000000000 -0500
@@ -21,11 +21,10 @@
 
 [ -z "$PULSE_SERVER" ]
 
-@PA_BINARY@ --start "$@"
-
 if [ x"$DISPLAY" != x ] ; then
 
     @PACTL_BINARY@ load-module module-x11-publish "display=$DISPLAY" > /dev/null
+    @PACTL_BINARY@ load-module module-x11-bell "display=$DISPLAY" "sample=bell.ogg" > /dev/null
     @PACTL_BINARY@ load-module module-x11-cork-request "display=$DISPLAY" > /dev/null
 
     if [ x"$SESSION_MANAGER" != x ] ; then
