Index: pulseaudio-0.9.19/src/daemon/start-pulseaudio-x11.in
===================================================================
--- pulseaudio-0.9.19.orig/src/daemon/start-pulseaudio-x11.in	2009-11-05 16:01:57.000000000 -0500
+++ pulseaudio-0.9.19/src/daemon/start-pulseaudio-x11.in	2009-11-05 16:04:14.000000000 -0500
@@ -26,6 +26,7 @@
 if [ x"$DISPLAY" != x ] ; then
 
     @PACTL_BINARY@ load-module module-x11-publish "display=$DISPLAY" > /dev/null
+    @PACTL_BINARY@ load-module module-x11-bell "display=$DISPLAY" "sample=bell.ogg" > /dev/null
     @PACTL_BINARY@ load-module module-x11-cork-request "display=$DISPLAY" > /dev/null
 
     if [ x"$SESSION_MANAGER" != x ] ; then
