Index: pulseaudio-0.9.21/src/pulsecore/cpu-arm.c
===================================================================
--- pulseaudio-0.9.21.orig/src/pulsecore/cpu-arm.c	2009-12-16 22:44:27.000000000 -0500
+++ pulseaudio-0.9.21/src/pulsecore/cpu-arm.c	2009-12-17 00:01:12.000000000 -0500
@@ -30,6 +30,7 @@
 #include <fcntl.h>
 
 #include <pulse/xmalloc.h>
+#include <pulsecore/core-util.h>
 #include <pulsecore/log.h>
 
 #include "cpu-arm.h"
@@ -67,7 +68,7 @@
         return NULL;
     }
 
-    if ((n = pa_read(fd, cpuinfo, MAX_BUFFER-1)) < 0) {
+    if ((n = pa_read(fd, cpuinfo, MAX_BUFFER-1, NULL)) < 0) {
         pa_xfree(cpuinfo);
         pa_close(fd);
         return NULL;
@@ -105,7 +106,8 @@
     }
     /* get the CPU features */
     if ((line = get_cpuinfo_line (cpuinfo, "Features"))) {
-        char *state = NULL, *current;
+        const char *state = NULL;
+        char *current;
 
         while ((current = pa_split_spaces (line, &state))) {
             if (!strcmp (current, "vfp"))
