Index: pulseaudio-0.9.22~0.9.21+341-g62bf/src/daemon/daemon.conf.in
===================================================================
--- pulseaudio-0.9.22~0.9.21+341-g62bf.orig/src/daemon/daemon.conf.in	2010-01-14 20:44:39.000000000 -0500
+++ pulseaudio-0.9.22~0.9.21+341-g62bf/src/daemon/daemon.conf.in	2010-01-14 20:44:44.000000000 -0500
@@ -55,7 +55,7 @@
 ; enable-remixing = yes
 ; enable-lfe-remixing = no
 
-; flat-volumes = yes
+flat-volumes = no
 
 ; rlimit-fsize = -1
 ; rlimit-data = -1
