Index: pulseaudio-0.9.14/src/modules/module-alsa-sink.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/module-alsa-sink.c	2009-02-20 01:50:11.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/module-alsa-sink.c	2009-02-20 01:50:40.000000000 -0500
@@ -94,8 +94,8 @@
 #define DEFAULT_DEVICE "default"
 #define DEFAULT_TSCHED_BUFFER_USEC (2*PA_USEC_PER_SEC)            /* 2s */
 #define DEFAULT_TSCHED_WATERMARK_USEC (20*PA_USEC_PER_MSEC)       /* 20ms */
-#define TSCHED_MIN_SLEEP_USEC (3*PA_USEC_PER_MSEC)                /* 3ms */
-#define TSCHED_MIN_WAKEUP_USEC (3*PA_USEC_PER_MSEC)               /* 3ms */
+#define TSCHED_MIN_SLEEP_USEC (10*PA_USEC_PER_MSEC)               /* 10ms */
+#define TSCHED_MIN_WAKEUP_USEC (4*PA_USEC_PER_MSEC)               /* 4ms */
 
 struct userdata {
     pa_core *core;
Index: pulseaudio-0.9.14/src/modules/module-alsa-source.c
===================================================================
--- pulseaudio-0.9.14.orig/src/modules/module-alsa-source.c	2009-02-20 01:50:11.000000000 -0500
+++ pulseaudio-0.9.14/src/modules/module-alsa-source.c	2009-02-20 01:51:10.000000000 -0500
@@ -95,8 +95,8 @@
 #define DEFAULT_DEVICE "default"
 #define DEFAULT_TSCHED_BUFFER_USEC (2*PA_USEC_PER_SEC)       /* 2s */
 #define DEFAULT_TSCHED_WATERMARK_USEC (20*PA_USEC_PER_MSEC)  /* 20ms */
-#define TSCHED_MIN_SLEEP_USEC (3*PA_USEC_PER_MSEC)           /* 3ms */
-#define TSCHED_MIN_WAKEUP_USEC (3*PA_USEC_PER_MSEC)          /* 3ms */
+#define TSCHED_MIN_SLEEP_USEC (10*PA_USEC_PER_MSEC)          /* 10ms */
+#define TSCHED_MIN_WAKEUP_USEC (4*PA_USEC_PER_MSEC)          /* 4ms */
 
 struct userdata {
     pa_core *core;
