Index: pulseaudio-0.9.22+stable-queue-18-geb966/src/Makefile.in
===================================================================
--- pulseaudio-0.9.22+stable-queue-18-geb966.orig/src/Makefile.in	2011-02-02 13:16:23.076223003 +0100
+++ pulseaudio-0.9.22+stable-queue-18-geb966/src/Makefile.in	2011-02-02 13:17:47.240223015 +0100
@@ -2656,7 +2656,10 @@
 		modules/alsa/mixer/paths/analog-input.conf \
 		modules/alsa/mixer/paths/analog-input.conf.common \
 		modules/alsa/mixer/paths/analog-input-fm.conf \
+		modules/alsa/mixer/paths/analog-input-dock-mic.conf \
+		modules/alsa/mixer/paths/analog-input-front-mic.conf \
 		modules/alsa/mixer/paths/analog-input-internal-mic.conf \
+		modules/alsa/mixer/paths/analog-input-rear-mic.conf \
 		modules/alsa/mixer/paths/analog-input-linein.conf \
 		modules/alsa/mixer/paths/analog-input-mic.conf \
 		modules/alsa/mixer/paths/analog-input-mic.conf.common \
