From: Ricardo Salveti de Araujo <rsalveti@rsalveti.net>
Date: Tue, 16 Sep 2014 18:00:59 -0300
Subject: [PATCH] module-stream-restore: use entry_write when filling up the  database

Signed-off-by: Ricardo Salveti de Araujo <rsalveti@rsalveti.net>
---
 src/modules/module-stream-restore.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/src/modules/module-stream-restore.c b/src/modules/module-stream-restore.c
index 23f4780..38a4680 100644
--- a/src/modules/module-stream-restore.c
+++ b/src/modules/module-stream-restore.c
@@ -1833,7 +1833,6 @@ static int fill_db(struct userdata *u, const char *filename) {
         *d = 0;
         if (pa_atod(v, &db) >= 0) {
             if (db <= 0.0) {
-                pa_datum key, data;
                 struct entry e;
 
                 pa_zero(e);
@@ -1842,13 +1841,7 @@ static int fill_db(struct userdata *u, const char *filename) {
                 pa_cvolume_set(&e.volume, 1, pa_sw_volume_from_dB(db));
                 pa_channel_map_init_mono(&e.channel_map);
 
-                key.data = (void *) ln;
-                key.size = strlen(ln);
-
-                data.data = (void *) &e;
-                data.size = sizeof(e);
-
-                if (pa_database_set(u->database, &key, &data, FALSE) == 0)
+                if (entry_write(u, ln, &e, FALSE))
                     pa_log_debug("Setting %s to %0.2f dB.", ln, db);
             } else
                 pa_log_warn("[%s:%u] Positive dB values are not allowed, not setting entry %s.", fn, n, ln);
-- 
2.1.0

