From ea0a2a603c970a5ae59a3b2a49ed189997c91ece Mon Sep 17 00:00:00 2001
From: Sean McNamara <smcnam@gmail.com>
Date: Fri, 25 Mar 2011 01:28:10 -0400
Subject: [PATCH 02/33] vala: delegate FreeCb does not have a target.

---
 vala/libpulse.vapi |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vala/libpulse.vapi b/vala/libpulse.vapi
index aed526a..8304911 100644
--- a/vala/libpulse.vapi
+++ b/vala/libpulse.vapi
@@ -49,7 +49,7 @@ namespace PulseAudio {
         [CCode (cname="PA_INVALID_INDEX")]
         public const uint32 INVALID_INDEX;
 
-        [CCode (cname="pa_free_cb_t")]
+        [CCode (cname="pa_free_cb_t", has_target=false)]
         public delegate void FreeCb(void *p);
 
         [CCode (cname="pa_sample_format_t", cprefix="PA_SAMPLE_")]
-- 
1.7.4.1

