From 9f52c1064651667f48877f54ae002e97eecd6289 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Sat, 26 Mar 2011 12:42:43 +0000
Subject: [PATCH 04/33] vala: Fix path error in the last commit.

---
 Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index c764f18..691f093 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -41,7 +41,7 @@ noinst_DATA =
 vapidir = $(datadir)/vala/vapi
 vapi_DATA = \
 		vala/libpulse.deps vala/libpulse.vapi \
-		libpulse-mainloop-glib.deps libpulse-mainloop-glib.vapi
+		vala/libpulse-mainloop-glib.deps vala/libpulse-mainloop-glib.vapi
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libpulse.pc libpulse-simple.pc
-- 
1.7.4.1

